packages/next/taskfile.js

```diff
@@ -1160,6 +1162,20 @@ export async function ncc_compression(task, opts) {
     .target('src/compiled/compression')
 }
 // eslint-disable-next-line camelcase
+externals['zod'] = 'next/dist/compiled/zod'
+export async function ncc_zod(task, opts) {
+  await task
+    .source(relative(__dirname, require.resolve('zod')))
+    .ncc({ packageName: 'zod', externals })
+    .target('src/compiled/zod')
+
+  // Copy all the .d.ts to the target directory
+  const zodPath = dirname(
+    relative(__dirname, require.resolve('zod/package.json'))
+  )
+  task.source(join(zodPath, '**/*.d.ts')).target(`src/compiled/zod`)
```

Comment:
Does it work if we do re-rexport in `misc.d.ts` instead of copying these types? 
```ts
declare module 'next/dist/compiled/zod' {
  import m from 'zod'
  export = m
}
```